crates/mun_mdbook_test_plugin/src/main.rs

```diff
@@ -0,0 +1,125 @@
+use mdbook::renderer::RenderContext;
+use mdbook::BookItem;
+
+use std::io;
+
+use regex::Regex;
+
+use mun_test::{CompileAndRunTestDriver, CompileTestDriver};
+
+use mun_runtime::{invoke_fn, RuntimeBuilder};
+
+use std::panic;
+
+enum TestingMethod {
+    Ignore,
+    CompileFail,
+    Compile,
+    CompileAndRun,
+}
+
+impl Default for TestingMethod {
+    fn default() -> Self {
+        TestingMethod::CompileAndRun
+    }
+}
+
+fn get_testing_method(code: &str) -> TestingMethod {
+    let mut testing_method = TestingMethod::default();
+
+    for line in code.lines() {
+        if let Some(first_char) = line.chars().next() {
+            if first_char != ',' {
+                return testing_method;
+            }
+        } else {
+            return testing_method;
+        }
+        if line == ",compile_fail" {
+            return TestingMethod::CompileFail;
+        }
+        if line == ",no_run" {
+            testing_method = TestingMethod::Compile;
+        }
+        if line == ",ignore" {
+            return TestingMethod::Ignore;
+        }
+    }
+
+    testing_method
+}
+
+fn test_code(code: &str) {
+    let testing_method = get_testing_method(code);
+
+    match testing_method {
```

Comment:
I'd remove the outer `match` statement altogether, and just use the inner match statement.